File: user-service/src/main/java/com/github/manosbatsis/services/user/service/UserService.java
package com.github.manosbatsis.services.user.service;

import com.github.manosbatsis.services.user.model.User;

import java.util.List;

public interface UserService {

    List<User> getUsers();

    User saveUser(User user);

    void deleteUser(User user);

    User validateAndGetUserById(Long id);

    void validateUserExistsByEmail(String email);
}

File: user-service/src/main/kotlin/com/github/manosbatsis/services/user/service/UserService.kt
package com.github.manosbatsis.services.user.service

import com.github.manosbatsis.services.user.model.User

interface UserService {

    fun findAll(): List<User>
    fun saveUser(user: User): User
    fun deleteUser(user: User)
    fun validateAndGetUserById(id: Long): User
    fun emailAvailableOrThrow(email: String)
}
